tests/runtime/pasqal/test_pasqal_runtime.py

```
@@ -0,0 +1,548 @@
+        assert payload["serialized_sequence"] == '{"already": "serialized"}'
+
+    def test_submit_empty_input_raises(self, mock_sdk):
+        provider = PasqalProvider(sdk=mock_sdk)
+        device = provider.get_device("EMU_FREE")
+        with pytest.raises(PasqalDeviceError, match="at least one"):
+            device.submit([])
+
+    def test_submit_invalid_type_raises(self, mock_sdk):
+        provider = PasqalProvider(sdk=mock_sdk)
+        device = provider.get_device("EMU_FREE")
+        with pytest.raises(PasqalDeviceError, match="pulser.Sequence"):
```

Ruff (0.15.13)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)